pkg/controller/pravega/pravega_segmentstore.go

```diff
@@ -44,7 +44,7 @@ func MakeSegmentStoreStatefulSet(pravegaCluster *api.PravegaCluster) *appsv1.Sta
 			Replicas:            &pravegaCluster.Spec.Pravega.SegmentStoreReplicas,
 			PodManagementPolicy: appsv1.OrderedReadyPodManagement,
 			UpdateStrategy: appsv1.StatefulSetUpdateStrategy{
-				Type: appsv1.RollingUpdateStatefulSetStrategyType,
+				Type: appsv1.OnDeleteStatefulSetStrategyType,
```

Comment:
Please update upgrade documentation (upgrade.md) to reflect that Segment Store upgrade strategy has been changed from "Rolling Update" to "OnDelete".

Reply:
done